src/tests/JIT/opt/And/IntAnd.cs

```diff
@@ -0,0 +1,171 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace CodeGenTests
+{
+    class IntAnd
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void SideEffect()
+        {
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static bool Test_UInt32_UInt32_And(uint x, uint y)
+        {
+            // X64-NOT: movzx
+
+            // We expect 'and reg8, reg8'.
+            // X64: and {{[a-z]+[l|b]}}, {{[a-z]+[l|b]}}
+
+            if ((byte)(x & y) == 0)
+            {
+                SideEffect();
```

Comment:
What is `SideEffect` doing here?  Do we get different code generation if we are only producing 1/0 vs needing control flow?

Reply:
I need to have some sort of side effect to happen. Because if I do this example:
```csharp
if ((byte)(x & y) == 0)
{
    return true;
}

return false;
```

I'm not going to get a branch. 

I need a branch in order to see the `and` op using 16-bit or 8-bit operations.